plate_mapper/plate_mapper.py

```diff
@@ -68,22 +96,62 @@ def plate_mapper(input_f, barseq_f, output_f):
     barseq_f.close()
     print('  Done.')
     # Write output sequencing run file
+    samples = {}
```

Comment:
Yes + clarity (you don't have to read the 20+ lines of code to do this), except that you could do:
`samples = [sample for ...]`
Then have other like this to write your file (this can be the last step if no warnings/errors), and summaries.